chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -930,3 +934,323 @@ async def set_nft_did(self, nft_coin_info: NFTCoinInfo, did_id: bytes, fee: uint
+                    launcher_coin.name(), metadata["program"], NFT_METADATA_UPDATER.get_tree_hash(), nft_inner_puzzle
+                )
+                new_nft_coin: Coin = Coin(eve_coin.name(), nft_fullpuz.get_tree_hash(), uint64(amount))
+                # Checking that the nft in the eve spend additions matches what we predict
+                assert nft_from_eve == new_nft_coin
+
+                # Uncurry the nft and update the state_layer so we can transfer it
+                unft = UncurriedNFT.uncurry(*eve_fullpuz.uncurry())
+                assert isinstance(unft, UncurriedNFT)  # mypy
+                nft_to_transfer = NFTCoinInfo(
+                    nft_id=launcher_coin.name(),
+                    coin=new_nft_coin,
+                    lineage_proof=LineageProof(
+                        parent_name=eve_coin.parent_coin_info,
+                        inner_puzzle_hash=unft.nft_state_layer.get_tree_hash(),
+                        amount=uint64(eve_coin.amount),
+                    ),
+                    full_puzzle=nft_fullpuz,
+                    mint_height=uint32(0),
+                )
+                transfer_tx = await self.generate_signed_transaction(
+                    [uint64(eve_coin.amount)],
+                    [target_ph],
+                    nft_coin=nft_to_transfer,
+                    new_owner=b"",
+                    new_did_inner_hash=b"",
+                )
+                transfer_spends.append(transfer_tx[0].spend_bundle)
+
+                #  Extract the Puzzle Announcement from the transfer spend
+                transfer_sb = transfer_tx[0].spend_bundle
+                assert isinstance(transfer_sb, SpendBundle)
+                nft_sol = transfer_sb.coin_spends[0].solution.to_program()
+                conds = nft_fullpuz.run(nft_sol)
+                puzzle_announcement = [x for x in conds.as_python() if int_from_bytes(x[0]) == 62][0][1]
+                assertion = std_hash(nft_fullpuz.get_tree_hash() + puzzle_announcement)
+                puzzle_assertions.add(assertion)
+
+        # We've now created all the intermediate, launcher, eve and transfer spends.
+        # Create the xch spend to fund the minting
+        spend_value = sum([coin.amount for coin in xch_coins])
+        change: uint64 = uint64(spend_value - total_amount)
+        xch_spends = []
+        if xch_change_ph is None:
+            xch_change_ph = await self.standard_wallet.get_new_puzzlehash()
+        xch_primaries = [
+            AmountWithPuzzlehash({"puzzlehash": xch_change_ph, "amount": change, "memos": [xch_change_ph]})
+        ]
+        for xch_coin in xch_coins:
+            puzzle: Program = await self.standard_wallet.puzzle_for_puzzle_hash(xch_coin.puzzle_hash)
+            solution: Program = self.standard_wallet.make_solution(
+                primaries=xch_primaries,
+                fee=fee,
+            )
+            xch_spends.append(CoinSpend(xch_coin, puzzle, solution))
```

Comment:
This loop seems like it shouldn't work which means you need to have a test where you select more than one xch_coin to do this operation.  In this loop, you're forcing every xch_coin to create every output and also it will pay the full fee for each.  You should probably be using the standard wallet's `generate_signed_transaction` it allows you to specify what coins you want to spend.

Reply:
We're passing in a specific xch coin that might not exist when we're creating the spend bundle (since it will be the change coin that will get created when an earlier spend bundle gets submitted. So the generate_signed_tx method isn't available and hence the "manual" solution. In general when bulk minting you'd set the fee to zero when creating spend bundles, and then add on a fee only when you're actually submitting the spends

Reply:
I think generate_signed_transaction will let you spend coins even that don't exist yet: https://github.com/Chia-Network/chia-blockchain/blob/7c9fb654a3563e69c54848d137859856c8566587/chia/wallet/wallet.py#L291.  But if for some reason that doesn't work, you still need to fix this loop because it will fail (thankfully) if more than one xch_coins are passed/selected.

Reply:
Actually it won't necessarily fail now that I think about it so the user might end up just spending len(xch_coins)x more money than expected